File: 504-Base-7.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        
        ans = ""
        num1 = num
        num = abs(num)
        while num > 0:
            ans = str(num % 7) + ans
            num //= 7

        return "-"+ans if num1 <0 else ans

File: 504-Base-7.java
class Solution {
    public String convertToBase7(int num) {
        if(num==0) return "0";

        String ans = "";
        int num_temp = num;
        num = Math.abs(num);
        while(num>0){
            ans = String.valueOf(num%7) + ans;
            num /= 7;
        }
        return num_temp<0 ? ("-"+ans) : ans;
    }
}